core/src/main/java/org/jdbi/v3/MappingRegistry.java

```diff
@@ -26,10 +27,10 @@
     private static final PrimitivesColumnMapperFactory BUILT_INS = new PrimitivesColumnMapperFactory();
 
     private final List<ResultSetMapperFactory> rowFactories = new CopyOnWriteArrayList<>();
-    private final ConcurrentHashMap<Class<?>, ResultSetMapper<?>> rowCache = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<Type, ResultSetMapper<?>> rowCache = new ConcurrentHashMap<>();
```

Comment:
From what I can see, all of the types included in the SDK implement equals using the public API of each type interface. Guava does this as well.